Packs/AggregatedCommands/Scripts/FileEnrichment/FileEnrichment.py

```diff
@@ -0,0 +1,916 @@
+
+    demisto.debug(f"Summarized results from all executed commands: {summaries}.")
+
+    return CommandResults(
+        outputs=merged_context,
+        readable_output=tableToMarkdown(name=f"File Enrichment result for {', '.join(file_hashes)}", t=summaries),
+    )
+
+
+def warn_about_invalid_args(
+    invalid_hashes: list[str], enrichment_brands: list[str], executed_brands: list[str]
+) -> CommandResults | None:
+    """
+    Returns a warning war room entry about invalid `file_hash` and `enrichment_brands` argument values, if found.
+
+    Args:
+        invalid_hashes (list[str]): List of file hashes where the type was classified as "UNKNOWN".
+        enrichment_brands (list[str]): List of brand names to run, as given in the `enrichment_brands` argument.
+        executed_brands (list[str]): List of brand names that returned execution results with context outputs.
+
+    Returns:
+        CommandResults | None: Warning command results if any invalid arguments, None otherwise.
+    """
+    warnings: dict[str, str] = {}
+    if invalid_hashes:
+        warnings["Skipped Hashes"] = ", ".join(invalid_hashes)
+
+    if skipped_brands := (set(enrichment_brands) - set(executed_brands)):
+        warnings["Skipped Brands"] = ", ".join(skipped_brands)
+
+    if not warnings:
+        return None
+
+    demisto.debug(f"Found {len(invalid_hashes)} invalid hashes and {len(skipped_brands)} invalid source brands.")
+    return CommandResults(readable_output=tableToMarkdown("Invalid inputs skipped", t=warnings), entry_type=EntryType.WARNING)
+
+
+""" SCRIPT FUNCTION """
+
+
+def file_enrichment_script(args: dict[str, Any]) -> list[CommandResults]:
+    """
+    Implements `!file-enrichment` script by searching for the File indicator in TIM and (optionally) running external enrichment.
+    It then outputs a human-readable summary of the command execution results and returns a consolidated merged `File` context.
+
+    Args:
+        args (dict): Script arguments.
+
+    Returns:
+        list[CommandResults]: List of CommandResults objects.
+
+    Raises:
+        ValueError: If the file hash is not valid.
+        DemistoException: If none of the enrichment brands has an enabled integration instance.
+    """
+    demisto.debug(f"Parsing and validating script args: {args}.")
+
+    file_hashes: list = argToList(args.get("file_hash", ""))
+    external_enrichment: bool = argToBoolean(args.get("external_enrichment", False))
+    enrichment_brands: list = argToList(args.get("enrichment_brands"))  # brands to use for external enrichment
+    verbose: bool = argToBoolean(args.get("verbose", False))
+    include_additional_fields: bool = argToBoolean(args.get("additional_fields", False))
+
+    # Classify hashes by type to check validity and handle commands that only support specific types
+    hashes_by_type: dict[str, list] = classify_hashes_by_type(file_hashes)
+    invalid_hashes = hashes_by_type.pop("UNKNOWN", [])
```

Comment:
Lets log the invalid hashes to prevent long debugging sessions with clients when they cannot find results for their hashes.